apps/server/src/modules/files/uc/sync-embedded-files.uc.ts

```diff
@@ -77,21 +115,33 @@ export class SyncEmbeddedFilesUc {
 		}
 	}
 
+	private async updateTaskLinks(file: SyncFileItem) {
+		const task = await this.embeddedFilesRepo.findTask(new ObjectId(file.parentId));
+
+		if (task) {
+			task.description = this.replaceLink(task.description, file);
+			await this.embeddedFilesRepo.updateTask(task);
+		}
+	}
+
 	private async updateLessonsLinks(file: SyncFileItem) {
 		const lesson = await this.embeddedFilesRepo.findLesson(new ObjectId(file.parentId));
 
 		if (lesson) {
 			lesson.contents = lesson.contents.map((item: IComponentProperties) => {
-				const regex = new RegExp(`${fileUrlRegex}${file.source.id}.+?"`, 'g');
-				const newUrl = `"/api/v3/file/download/${file.fileRecord.id}/${file.fileRecord.name}"`;
-
 				if ('text' in item.content) {
-					item.content.text = item.content.text.replace(regex, newUrl);
+					item.content.text = this.replaceLink(item.content.text, file);
 				}
 
 				return item;
 			});
 			await this.embeddedFilesRepo.updateLesson(lesson);
 		}
 	}
+
+	private replaceLink(text: string, file: SyncFileItem) {
+		const regex = new RegExp(`${fileUrlRegex}${file.source.id}.+?"`, 'g');
+		const newUrl = `"/api/v3/file/download/${file.fileRecord.id}/${file.fileRecord.name}"`;
```

Comment:
This work for legacy client if a endpoint exist in it.
But do it also work in future for vue single page browser application? Can we add a endpoint over vue router that match?
Do it work for the CkEditor that the content is displayed?

Reply:
The hostname is always the same and the server (as a proxy) forwards it. It will (and should) work!